### ISSUE TITLE
Removing subject_alternative_names parameter doesn't force a new certificate

### ISSUE THREAD

**Issue by tomelliff** [terraform-providers/terraform-provider-aws#8755]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.14
+ provider.aws v2.11.0

### Affected Resource(s)

* aws_acm_certificate

Removing `subject_alternative_names` from `aws_acm_certificate` resource doesn't force a new certificate to be created.

### Terraform Configuration Files

```hcl
resource "aws_acm_certificate" "certificate" {
  domain_name = "foo.example.com"

  subject_alternative_names = [
    "bar.example.com",
    "baz.example.com",
  ]

  validation_method = "DNS"
}
```

Then remove the `subject_alternative_names` parameter:

```hcl
resource "aws_acm_certificate" "certificate" {
  domain_name = "foo.example.com"

  validation_method = "DNS"
}
```

And run a plan to see that Terraform doesn't want to do anything.

### Expected Behavior

Should want to recreate the certificate without any SANs.

### Actual Behavior

Terraform plan shows no changes.

### Steps to Reproduce

1. `terraform apply` first config
2. `terraform plan` second config

**Comment by dekimsey** [comment]
I'm also running into the reverse situation. Adding a `subject_alternative_names = ['foo.com']` attribute to my _existing_ aws_acm_certificate resource does not trigger a change. (And apparently nor does tainting it!) So I'm unable to add the validation record resources.

```
Error: Invalid index

  on example.tf line 10, in resource "aws_route53_record" "example_san1":
 178:   name    = aws_acm_certificate.example.domain_validation_options.1.resource_record_name
    |----------------
    | aws_acm_certificate.example.domain_validation_options is list of object with 1 element

The given key does not identify an element in this collection value.
```

**Comment by bflad** [comment]
A documentation update has been merged and will deploy when version 2.65.0 of the Terraform AWS Provider is released, likely later today. 👍
